data_steward/tools/copy_dataset_to_output_prod.py

```diff
@@ -0,0 +1,204 @@
+    """
+    if tier.lower() not in TIER_LIST:
+        msg = f"Parameter ERROR: {tier} is an incorrect input for the tier parameter, accepted: controlled or " \
+              f"registered"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def validate_deid_stage_param(deid_stage):
+    """
+    helper function to validate the deid_stage parameter passed is correct, must be 'deid', 'base' or 'clean'
+
+    :param deid_stage: deid_stage parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if deid_stage not in DEID_STAGE_LIST:
+        msg = f"Parameter ERROR: {deid_stage} is an incorrect input for the deid_stage parameter, accepted: deid, " \
+              f"base, clean"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+
+
+def validate_release_tag_param(arg_value):
+    """
+    User defined helper function to validate that the release_tag parameter follows the correct naming convention
+
+    :param arg_value: release tag parameter passed through either the command line arguments
+    :return: arg_value
+    """
+
+    release_tag_regex = re.compile(r'[0-9]{4}Q[0-9]R[0-9]')
+    if not re.match(release_tag_regex, arg_value):
+        msg = f"Parameter ERROR {arg_value} is in an incorrect format, accepted: YYYYQ#R#"
+        LOGGER.error(msg)
+        raise argparse.ArgumentTypeError(msg)
+    return arg_value
+
+
+def validate_create_tier_args(tier, stage, tag):
+    """
+    User defined helper function to validate that the tier, deid_stage, release_tag parameter
+     follows the correct naming convention
+    :param tier: tier parameter passed through from either a list or command line argument
+    :param stage: deid_stage parameter passed through from either a list or command line argument
+    :param tag: release tag parameter passed through either the command line arguments
+    :return: None, breaks if not valid params are passed
+    """
+    validate_tier_param(tier)
+    validate_deid_stage_param(stage)
+    validate_release_tag_param(tag)
+
+
+def get_dataset_name(tier, release_tag, deid_stage):
+    """
+    Helper function to create the output prod dataset name based on the given criteria
+    This function should return a name for the final dataset only (not all steps along the way)
+    The function returns a string in the form: [C|R]{release_tag}[_base|_clean]
+
+    :param tier: controlled or registered tier intended for the output dataset
+    :param release_tag: release tag for dataset in the format of YYYYQ#R#
+    :param deid_stage: deid stage (deid, base or clean)
+    :return: a string for the dataset name
+    """
+    # validate parameters
+    validate_create_tier_args(tier, deid_stage, release_tag)
+
+    tier = tier[0].upper()
+    release_tag = release_tag.upper()
+    deid_stage = '' if release_tag == 'deid' else f'_{deid_stage}'
+
+    dataset_name = f"{tier}{release_tag}{deid_stage}"
```

Comment:
Another quirk about the output-prod project: The clean dataset won't have `_clean`, it will be just the tier and release tag. The base dataset will have `_base`

Reply:
Sounds good, will work on all these suggestions.